addons/mail/static/src/messaging/entity/entity_core.js

```diff
@@ -0,0 +1,246 @@
+ * @throws {Error} in case some relations are not correct
+ */
+function checkRelations(Entities) {
+    for (const Entity of Object.values(Entities)) {
+        for (const relationName in Entity.relations) {
+            const relation = Entity.relations[relationName];
+            if (!relation.type) {
+                throw new Error(
+                    `Relation "${Entity.name}/${relationName}" must define a type in "type".`
+                );
+            }
+            if (!(['one2one', 'one2many', 'many2one', 'many2many'].includes(relation.type))) {
+                throw new Error(
+                    `Relation "${Entity.name}/${relationName}" has invalid type "${relation.type}".`
+                );
+            }
+            if (!relation.inverse) {
+                throw new Error(
+                    `Relation "${
+                        Entity.name
+                    }/${
+                        relationName
+                    }" must define an inverse relation name in "inverse".`
+                );
+            }
+            if (!relation.to) {
+                throw new Error(
+                    `Relation "${
+                        Entity.name
+                    }/${
+                        relationName
+                    }" must define an Entity class name in "to".`
+                );
+            }
+            const RelatedEntity = Entities[relation.to];
+            if (!RelatedEntity) {
+                throw new Error(
+                    `Entity class name of relation "${Entity.name}/${relationName}" does not exist.`
+                );
+            }
+            const inverseRelation = RelatedEntity.relations[relation.inverse];
+            if (!inverseRelation) {
+                throw new Error(
+                    `Relation entity class "${
+                        Entity.name
+                    }/${
+                        relationName
+                    }" has no inverse relation "${RelatedEntity.name}/${relation.inverse}".`
+                );
+            }
+            const allSelfAndParentNames = [];
+            let target = Entity;
+            while (target) {
+                allSelfAndParentNames.push(target.name);
+                target = target.__proto__;
+            }
+            if (!allSelfAndParentNames.includes(inverseRelation.to)) {
+                throw new Error(
+                    `Relation "${
+                        Entity.name
+                    }/${
+                        relationName
+                    }" has inverse relation "${
+                        RelatedEntity.name
+                    }/${
+                        relation.inverse
+                    }" misconfigured (currently "${
+                        inverseRelation.to
+                    }", should instead refer to this entity or parented entity: ${
+                        allSelfAndParentNames.map(name => `"${name}"`).join(', ')
+                    }?)`
```

Comment:
It's really hard to read the sentence, maybe we should use sprintf in those cases?